buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchNode.java

```diff
@@ -782,6 +782,13 @@ private Map<String, String> getOpenSearchEnvironment() {
         defaultEnv.put("HOSTNAME", HOSTNAME_OVERRIDE);
         defaultEnv.put("COMPUTERNAME", COMPUTERNAME_OVERRIDE);
 
+        // Pass PATH environment variable to be able to run OpenSearch cli tools
```

Comment:
I believe we may not be fixing the right place, the shell scripts `./bin/opensearch-plugin` should handle that gracefully? (since this is where the failure happens)

Reply:
When `opensearch-plugin` script is executed it does not have any information about an environment of its parent process as the environment was cleared.

I think we should preserve all the environment like a shell do it for us when we run programs.

Reply:
> I think we should preserve all the environment like a shell do it for us when we run programs.

Correct, sorry if it was not clear, I meant not messing up with path but check if `dirname` is available, and if not use alternative means ?

Reply:
By default `Exec` task keeps the environment if we do not provide ours:

```
The environment variables to use for the process. Defaults to the environment of this process.
```

When we call a `setEnvironment` method we should merge our environment with the environment of the process (if we need it of cource).